### PR TITLE
[MIG][11.0] Migrate barcodes_generator_product

### PR DESCRIPTION
**Pull request by phatnguyenuit**
Depends on:

- [x] barcodes_generator_abstract - #122

### CODE REVIEW

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Hi @phatnguyenuit. thanks for porting this module !

For the time being this module depends on a unported module _abstract. If you want to make travis green, you should make this PR depending on https://github.com/OCA/stock-logistics-barcode/pull/122
Otherwise, just wait #122 is merged and rebuild.

**Comment by misern2** [comment]
@phatnguyenuit the dependcy is already merged in #138.
Could you please rebase ?

File: barcodes_generator_product/__manifest__.py [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 {
     'name': 'Generate Barcodes for Products',
     'summary': 'Generate Barcodes for Products (Templates and Variants)',
-    'version': '10.0.1.0.0',
+    'version': '11.0.1.0.0',
     'category': 'Tools',
     'author':
         'GRAP,'
         'La Louve,'
+        'Komit,'
```
Comment (legalsylvain) [inline barcodes_generator_product/__manifest__.py]: please remove Komit as author.
thanks.

**Review by legalsylvain — CHANGES_REQUESTED** [review]

File: barcodes_generator_product/models/barcode_rule.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Copyright (C) 2014-Today GRAP (http://www.grap.coop)
+# Copyright (C) 2016-Today La Louve (http://www.lalouve.net)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields
```
Comment (hieulucky111) [inline barcodes_generator_product/models/barcode_rule.py]: Import Order!

File: barcodes_generator_product/models/product_template.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        related='product_variant_ids.barcode_rule_id.generate_type')
+
+    # View Section
+    @api.multi
+    def generate_base(self):
+        self.product_variant_ids.generate_base()
+
+    @api.multi
+    def generate_barcode(self):
+        self.product_variant_ids.generate_barcode()
```
Comment (hieulucky111) [inline barcodes_generator_product/models/product_template.py]: Please add `ensure_one()` or use loop in here.

**Review by hieulucky111 — CHANGES_REQUESTED** [review]
Please check my review.

**Review by misern2 — APPROVED** [review]
Code reviewed and functional testing. LGTM 👍

**Review by legalsylvain — APPROVED** [review]
thanks.

**Comment by legalsylvain** [comment]
Merging, as all remarks have been take into account.

**Pull request merged**